extlinks/aggregates/management/commands/reaggregate_link_archives.py

```diff
@@ -0,0 +1,488 @@
+            )
+            > 0
+        )
+
+    def _has_aggregates_for_day(
+        self, existing_link_aggregates_in_object_storage, day_to_fix
+    ):
+        """
+        This function checks whether there are existing aggregates for the day to fix.
+        Parameters
+        ----------
+        existing_link_aggregates_in_object_storage :  An array of existing link aggregates from object storage.
+
+        day_to_fix :  str
+
+        Returns
+        -------
+        bool: whether there are existing aggregates for a given day in object storage
+        """
+        day_to_fix_formatted = (
+            datetime.fromisoformat(day_to_fix).date().strftime("%Y-%m-%d")
+        )
+        return (
+            len(
+                [
+                    i
+                    for i in existing_link_aggregates_in_object_storage
+                    if day_to_fix_formatted in i
+                ]
+            )
+            > 0
+        )
+
+    def _has_link_events_for_month(self, first_day_of_month, last_day_of_month):
+        return LinkEvent.objects.filter(timestamp__gte=first_day_of_month, timestamp__lte=last_day_of_month).count() > 0
+
+    def _has_link_events_for_day(self, day_to_fix):
+        day = datetime.fromisoformat(day_to_fix)
+        return LinkEvent.objects.filter(timestamp__gte=day, timestamp__lte=day + timedelta(days=1)).count() > 0
+
+
+    def _process_daily_aggregates(
+        self, collections, day_to_fix, directory, url_patterns
+    ):
+        """
+        This function loops through each url pattern and link event to fill the daily aggregates.
+        Parameters
+        ----------
+        collections :  An array of collections
+
+        day_to_fix :  str
+
+        directory :  str
+
+        url_patterns :  An array of url patterns
+
+        Returns
+        -------
+        None
+        """
+        # pull month string from day input parameter
+        month_to_fix = day_to_fix[:-2]
+        # load and split link events by url pattern
+        events_split_by_url_pattern = self._load_events_from_archives(
+            directory, month_to_fix, [i.url for i in url_patterns]
+        )
+        # loop through each collection
+        for collection in collections:
+            collection_url_pattern_strings = [
+                i.url for i in url_patterns.filter(collection=collection)
+            ]
+            # loop through each collection's URLPatterns
+            for collection_url_string in collection_url_pattern_strings:
+                # get the link events for the collection and day
+                for link_event in self._get_link_events_for_day(
+                    collection_url_string,
+                    events_split_by_url_pattern,
+                    int(day_to_fix[-2:]),
+                ):
+                    # create daily aggregates
+                    self._fill_daily_aggregate(collection, link_event)
+
+    def _fill_daily_aggregate(self, collection, link_event):
```

Comment:
Could we call the `fill_link_aggregates`, `fill_user_aggregates`, and `fill_pageproject_aggregates` commands here? I notice you are only filling one of the 3 aggregate tables, and you need to fill all 3.

Reply:
> Unless you think it's worth refactoring the existing aggregation commands to be usable by the backfill command?

I'd leave that up to you. It might be a bigger lift to do that instead of just adding the UserAggregates and PageProjectAggregates in this command. Whichever option you choose, I'll support.

> This also makes me feel like we shouldn't archive link events unless we're certain that they have been aggregated. 

I think we already check for that, but it's not collection-specific AFAIK. We can check the archive command to be sure this is the case